internal/ethapi/api.go

```diff
@@ -1889,6 +1889,36 @@ func (api *PrivateDebugAPI) SetHead(number hexutil.Uint64) {
 	api.b.SetHead(uint64(number))
 }
 
+// Burned returns the amount of burned ether for a specific
+// range of blocks under eip-1559.
+// The returned value is in denoted in Wei.
+func (api *PrivateDebugAPI) Burned(start, end *hexutil.Uint64) (*hexutil.Big, error) {
+	burned := big.NewInt(0)
+	startBlock := 0
+	endBlock := int(api.b.CurrentHeader().Number.Int64())
+	if start != nil {
+		startBlock = int(*start)
+	}
+	if end != nil {
+		endBlock = int(*end)
+	}
+	if startBlock > endBlock {
+		return (*hexutil.Big)(burned), errors.New("invalid range specified, start > end")
+	}
+	for i := startBlock; i < endBlock; i++ {
+		header, err := api.b.HeaderByNumber(context.Background(), rpc.BlockNumber(i))
+		if err != nil {
+			return (*hexutil.Big)(burned), err
+		}
+
+		if basefee := header.BaseFee; basefee != nil && basefee.Cmp(common.Big0) != 0 {
+			burnedBlock := new(big.Int).Mul(basefee, big.NewInt(int64(header.GasUsed)))
+			burned = new(big.Int).Add(burned, burnedBlock)
```

Comment:
Simpler and two less allocs:
```suggestion
			gasUsed := big.NewInt(int64(header.GasUsed))
			burned.Add(burned, gasUsed.Mul(gasUsed, basefee ))
```
?